### PR TITLE
script to generate all single base substitutions for a given species/chromosome/gene

### PR DESCRIPTION
**Pull request by ima23**

### CODE REVIEW

File: tools/variant_simulator/README.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+=====================================
+The script generates all single base substitutions (SNPs) for protein_coding gene(s) given a specific species or chromosome or gene. One can restrict the SNPs to be generated only for the introns, exons or only coding exons and a specific number of bases around each of them.
+
+##### Table of contents
+* [Usage](#usage)
+* [Examples](#examples)
+---
+<a name="usage"></a>
+### Usage
+The script generates all single base substitutions for protein coding genes given a specific species or chromosome or gene. The script uses the Esembl API and Ensembl databases.
```
Comment (sarahhunt) [inline tools/variant_simulator/README.md]: Ensembl!

File: tools/variant_simulator/simulate_variation.pl [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+    }
+  }
+}
+
+sub usage {
+
+  print qq{
+  Usage: perl simulate_variation.pl [OPTIONS]
+
+  Generate a VCFv4 file with all possibel SNPs for protein_coding genes given specific species or chromosome or gene.
```
Comment (sarahhunt) [inline tools/variant_simulator/simulate_variation.pl]: possible

File: tools/variant_simulator/simulate_variation.pl [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+
+  Options:
+
+      -registry          Ensembl registry file containing database connection details (Required)
+      -species           Progress information is printed (Default: human)
```
Comment (sarahhunt) [inline tools/variant_simulator/simulate_variation.pl]: This is an odd comment.

File: tools/variant_simulator/simulate_variation.pl [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+
+      -output|o          Output file (Default: simulated.vcf)
+
+      -chrom             Generate SNPs only for specified chromosome
+      -gene              Generate all SNPs for specified gene
+      -codingOnly        Generate SNPs only for translatable exons of protein_coding transcripts
+      -exonsOnly         Generate SNPs only for all exons of the genes
+      -intronsOnly       Generate SNPs only for all introns of the genes
+
+      -edge              Upstream/downstream bp of each feature (Default: 0)
```
Comment (sarahhunt) [inline tools/variant_simulator/simulate_variation.pl]: Can this description be clarified? 'Length of flanking region in which to generate SNPs' ?

**Comment by sarahhunt** [comment]
@ima23 - looks good! I just spotted a couple of typos.

**Comment by ima23** [comment]
@sarahhunt thanks for checking and spotting them! Fixed in next commit.

**Comment by sarahhunt** [comment]
Thanks @ima23 ! 
It runs quickly for a single gene. It would be helpful if specifying a registry file was optional and the public databases were used by default. Also, VCFtools is complaining about the info formating.

**Pull request merged**